### PR TITLE
Group sections and add table of contents in USAGE.md

### PR DESCRIPTION
**Pull request by Xymph**
Commits speak for themselves, I guess.

### CODE REVIEW

**Comment by Xymph** [comment]
One from the afterthought department: that setUserAgent comment would be a more [effective reminder](https://github.com/hamstar/Wikimate/pull/107#discussion_r664447951) if it's also in the Usage section?

File: USAGE.md [hunk 1]
```diff
@@ -228,7 +245,7 @@ if ($file->exists()) die('Example image already exists');
 $result = uploadFromUrl('https://upload.wikimedia.org/wikipedia/en/a/a9/Example.jpg', 'Adopt Wiki example image');
 ```
 
-### History...
+#### History...
```
Comment (waldyrious) [inline USAGE.md:248]: How about "Viewing the history", "Browsing history", or something along those lines?

And maybe the same for the "Other stuff" subsections: "Running custom queries", "Customizing the user agent", "Handling errors".

What do you think?

**Comment by waldyrious** [comment]
> Commits speak for themselves, I guess.

IMO the message of the second commit should make a reference to the USAGE doc, just like the first one does. I know that we're grouping commit series together with merge commits, but it's best to reduce implicit content wherever possible.

> One from the afterthought department: that [setUserAgent comment](https://github.com/hamstar/Wikimate/commit/f6983f739f17a9b05de1dbb695578f2321038484#diff-915f1e0f55c2400eb73093b0452dbac786cbcb0a04a0f5b0db0e4adaf5b81fbbR205) would be a more [effective reminder](https://github.com/hamstar/Wikimate/pull/107#discussion_r664447951) if it's also in the Usage section?

Definitely. And this PR seems like a good place to add that change :)

**Comment by waldyrious** [comment]
By the way, let's make sure to always use the [imperative mood](https://chris.beams.io/posts/git-commit/#imperative) in commit messages and PR titles. That's already the general practice, but a few exceptions (like those in this PR) have slipped by.

File: USAGE.md (hunk 1 above)
Comment (Xymph) [reply USAGE.md]: Good suggestions, adopted most of them.

**Comment by Xymph** [comment]
> IMO the message of the second commit should make a reference to the USAGE doc, just like the first one does. I know that we're grouping commit series together with merge commits, but it's best to reduce implicit content wherever possible.

Hmm, no original commit message did, but I can add it to both?

> > One from the afterthought department: that [setUserAgent comment](https://github.com/hamstar/Wikimate/commit/f6983f739f17a9b05de1dbb695578f2321038484#diff-915f1e0f55c2400eb73093b0452dbac786cbcb0a04a0f5b0db0e4adaf5b81fbbR205) would be a more [effective reminder](https://github.com/hamstar/Wikimate/pull/107#discussion_r664447951) if it's also in the Usage section?
> 
> Definitely. And this PR seems like a good place to add that change :)

Added as separate commit 2045aa9.

File: USAGE.md [hunk 2]
```diff
@@ -339,3 +348,20 @@ $useragent = $wiki->getUserAgent();
 $wiki->setUserAgent('Custom User Agent');
 $wiki->setUserAgent('Custom Prefix - ' . $useragent);
 ```
+
+In order to use a custom user agent for all requests in the session,
+call this method before invoking Wikimate::login().
```
Comment (waldyrious) [inline USAGE.md:353]: Shouldn't this be formatted as code?
```suggestion
call this method before invoking `Wikimate::login()`.
```

**Comment by waldyrious** [comment]
> Hmm, no original commit message did, but I can add it to both?

Oh, right, I was confused by github's weird habit of showing the PR title as the main title (and the html `<title>` when viewing a PR's commit, like [this](https://github.com/hamstar/Wikimate/pull/108/commits/8d0c91ff82c4fef96e3ed46a09434df064a0e64c). Anyway, yeah, please add the context to all commit messages so they can be read independently from the other ones.

> How does everything look now?

Should be ready after the commit messages are adjusted, and the inline comment I just added is addressed.

File: USAGE.md (hunk 2 above)
Comment (Xymph) [reply USAGE.md]: Naturally :roll_eyes: Thx.

**Comment by Xymph** [comment]
> Should be ready after the commit messages are adjusted, and the inline comment I just added is addressed.

New round pushed.

**Review by waldyrious — APPROVED** [review]
LGTM, nice work!

**Pull request merged**